.github/workflows/index-summary-visualizer.yaml

```diff
@@ -0,0 +1,75 @@
+name: Index and cleanup gh-pages reports
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 3 * * *" # Daily at 03:00 UTC
+
+jobs:
+  cleanup-and-index:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout gh-pages
+        uses: actions/checkout@v5
+        with:
+          ref: gh-pages
+
+      - name: Cleanup old reports and regenerate index
+        run: |
+          set -euo pipefail
+
+          # Find reports excluding index.html
+          reports=$(ls index.*.html 2>/dev/null | grep -v '^index\.html$' || true)
+
+          if [ -z "$reports" ]; then
+            echo "No reports found, nothing to do."
+            exit 0
+          fi
+
+          # Sort reports by filename (newest first)
+          sorted_reports=$(echo "$reports" | sort -r)
+
+          # Keep only the first 128
+          keep=$(echo "$sorted_reports" | head -n 128)
+          remove=$(comm -23 <(echo "$sorted_reports" | sort) <(echo "$keep" | sort))
+
+          # Remove old ones
+          if [ -n "$remove" ]; then
+            echo "Removing old reports:"
+            echo "$remove"
+            rm "$remove"
+          fi
+
+          # Regenerate index.html
+          echo "Generating index.html"
```

Comment:
I think this is fine, but just flagging that github pages makes it pretty easy to render markdown -- this could be a markdown list: https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll